### DISCUSSION TITLE
application-web userActionAndSessionProperties uniqueId Question

### DISCUSSION THREAD

**Discussion by agardnerIT** [Dynatrace/dynatrace-configuration-as-code#790] [Q&A]
Hi team,

When pushing an `application-web` which contains user actions properties, objects within the `userActionAndSessionProperties` array require a field called `uniqueId`.

Dynatrace API describes this field as: `Unique id among all userTags and properties of this application`

![image](https://user-images.githubusercontent.com/26523841/211450816-e4e612d6-257e-477d-83ba-35a180b4defa.png)

Is it possible for monaco to work with this? Can monaco dynamically look up the last known ID and increment it?

I appreciate this isn't a monaco issue but when an application has previously had tags - but it no longer does - I get this error from DT:

```
{"error":{"code":400,"message":"Tag with UniqueId 3 was deleted before"}}, responsible config: monaco_config/application-web/OpenFeature.json
```

Meaning `3` can apparently never be used again.

**Comment by UnseenWizzard** [comment]
Hi @agardnerIT no, this is not something monaco can currently work with - and to even consider adding it, we'd need a good way to get an application's latest available unique ID. That seems impossible as is, given that the property might have been deleted and thus not even contained in the JSON payload of an existing application anymore. 

As you already point out this appears to be more a (strange to me) API constraint - especially not being able to reuse/recreate properties. Could you please bring this up as a question about that API internally instead? There might be a good reason for this, or there might be a chance to adapt the API constraints

**Comment by ibarkowski** [comment]
Hi, I have the same problem. Additionally regarding to userActionAndSessionProperties also key of the property is checked. Monaco fails if the key existed before.
